unison-share-api/src/Unison/Server/NameSearch.hs

```diff
@@ -0,0 +1,78 @@
+module Unison.Server.NameSearch where
```

Comment:
This module pulls out the NameSearch and Search abstractions into their own module;

Previously there was only one implementation for them, but now we offer a Sqlite implementation alongside the existing 'NamesWithHistory' based version, so I've split it up  into modules.

This module contains the interface, the existing implementation is in `NameSearch.FromNames` and the new one in `NameSearch.Sqlite`

This module contains